discovery/resolver.go

```diff
@@ -69,4 +78,6 @@ func (r *watcherResolver) Close() {}
 // "ResolveNow will be called by gRPC to try to resolve the target name
 // again. It's just a hint, resolver can ignore this if it's not necessary.
 // It could be called multiple times concurrently."
-func (r *watcherResolver) ResolveNow(_ resolver.ResolveNowOptions) {}
+func (r *watcherResolver) ResolveNow(_ resolver.ResolveNowOptions) {
+	_ = r.cc.WaitForTransition(context.Background(), r.addr)
```

Comment:
This currently just waits until timeout because the `clientConnWrapper` instance created in `watcherResolver.Build` is a different one than the wrapper created in `watcherBalancerBuilder.Build`